miro_preprocessor/miro_copy_s3_asset/miro_copy_s3_derivative_asset/src/miro_copy_s3_derivative_asset.py

```diff
@@ -0,0 +1,50 @@
+from functools import partial
+import json
+import os
+
+import boto3
+
+from wellcome_lambda_utils.miro_utils import MiroImage
+from wellcome_lambda_utils import sns_utils
+from wellcome_lambda_utils.s3_utils import S3_Identifier
+from wellcome_lambda_utils import s3_utils
+
+
+def copy_and_forward_message(s3_client,
+                             sns_client,
+                             image_info,
+                             topic_arn,
+                             source_identifier,
+                             destination_identifier,
+                             source_head_response):
+    s3_utils.copy_asset_if_not_exists(s3_client=s3_client,
+                                      source_head_response=source_head_response,
```

Comment:
Passing this around seems decidedly iffy to me; is it really correct to be passing around S3 responses between our functions? Seems like that would make it very hard to test.

I might have missed this function going into the s3_utils module – I think a neater way to do this would be to have `s3_utils.copyfile(src, dst, lazy=False)`. Then the work of looking up this HEAD response is owned entirely by the module, and not exposed to the caller.

Reply:
👍  I'll use that from #1086 when it's ready